pic32/libraries/Stepper/Stepper.cpp

```diff
@@ -112,7 +114,13 @@ Stepper::Stepper(int number_of_steps, int motor_pin_1, int motor_pin_2, int moto
 */
 void Stepper::setSpeed(long whatSpeed)
 {
-  this->step_delay = 60L * 1000L / this->number_of_steps / whatSpeed;
+  // avoid divide by 0
+  if (whatSpeed > 0) {
+    this->step_delay = 60L * 1000L / this->number_of_steps / whatSpeed;
+  } else {
+    // divide by 0 attempted, set speed to minimum rpms
+    this->step_delay = -1; 
```

Comment:
If whatSpeed is passed in as 0, step_delay will be set to -1. On line 142, when we check to see if step_delay has passed, the if() will never fire, because step_delay is negative, and steps_left will never decrement, and thus we'll never leave the step() function and the sketch will hang. 

I think. :-)

Maybe add a check on line 140 that sees if step_delay is 0 or -1, and if so, it just returns from step()?